src/mq.c

```diff
@@ -561,53 +522,48 @@ amqp_bytes_t mq_declare_new_queue(const char *name)
 	if (amqp_get_rpc_reply(mq_ctx.conn).reply_type !=
 			       AMQP_RESPONSE_NORMAL) {
 		l_error("Error declaring queue name");
-		queue.bytes = NULL;
-		return queue;
+		worker_queue.bytes = NULL;
+		return -1;
 	}
 
-	queue = amqp_bytes_malloc_dup(r->queue);
-	if (queue.bytes == NULL)
+	worker_queue = amqp_bytes_malloc_dup(r->queue);
+	if (worker_queue.bytes == NULL)
 		l_error("Out of memory while copying queue buffer");
-
-	return queue;
+	return 1;
 }
 
 /**
  * mq_delete_queue:
- * @name: queue name
  *
- * Delete a queue in amqp connection.
+ * Delete the current queue in amqp connection.
  *
- * Returns: 0 if successful and -1 otherwise.
  */
-int mq_delete_queue(amqp_bytes_t queue)
+void mq_delete_queue()
 {
-	if (!mq_ctx.conn)
-		return -1;
-
-	amqp_queue_delete(mq_ctx.conn, 1, queue, 0, 0);
-	if (amqp_get_rpc_reply(mq_ctx.conn).reply_type !=
-			       AMQP_RESPONSE_NORMAL) {
-		l_error("Error deleting queue name");
-		return -1;
+	if (worker_queue.bytes) {
+			if (mq_ctx.conn) {
+				amqp_queue_delete(mq_ctx.conn, 1, worker_queue, 0, 0);
+				if (amqp_get_rpc_reply(mq_ctx.conn).reply_type !=
+							AMQP_RESPONSE_NORMAL) {
+					l_error("Error deleting queue name");
+				}
+			}
+		amqp_bytes_free(worker_queue);
+		worker_queue = amqp_empty_bytes;
 	}
-
```

Comment:
Is removing this line necessary?

Reply:
Yes, otherwise there will be two blank lines

Reply:
Yes, because the return statement was removed, so this is the method's last line (a curly bracket)